crates/forge_loader/src/manifest.rs

```diff
@@ -25,24 +25,54 @@ pub struct FunctionMod<'a> {
     providers: Option<AuthProviders<'a>>,
 }
 
+// Abstracting away key, function, and resolver into a single struct for reuse whoo!
 #[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
-struct ModInfo<'a> {
+struct CommonKey<'a> {
```

Comment:
You might want to add a helper method for getting the functions of this struct, since you'll be calling it a lot. One approach would
be:

```rust
impl<'a> CommonKey<'a> {
    fn append_functions<I: Extend<Item = &'a str> + ?Sized>(&self, funcs: &mut I) {
        funcs.extend(self.key);
        if let Some(Resolver { function }) = self.resolver {
            funcs.extend(function);
        }
    }
}
```

So that you can use `module.common.append_functions(&mut invokable_functions);` for each of the invokable functions. That way, you're less likely to forget to potentially add either the `function` or `resolver.function` property.